js-envs/src/main/scala/org/scalajs/jsenv/AsyncJSRunner.scala

```diff
@@ -38,12 +38,25 @@ trait AsyncJSRunner {
 
   /** Await completion of the started Run for the duration specified
    *  by [[atMost]]. Strictly equivalent to:
-   * 
+   *
    *  {{{
    *  Await.result(future, atMost)
    *  }}}
-   * 
+   *
    */
   final def await(atMost: Duration): Unit = Await.result(future, atMost)
 
+  /** Await completion of the started Run for the duration specified by
+   *  [[atMost]], or force it to stop if the timeout expires.
+   */
+  final def awaitOrStop(atMost: Duration): Unit = {
+    try {
+      await(atMost)
+    } catch {
+      case th: Throwable =>
```

Comment:
No we also need to stop the vm on any other failure. The point of this method is: I want you to stop, no matter what, but I give you a chance to do so gracefully first.